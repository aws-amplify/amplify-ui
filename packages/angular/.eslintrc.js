module.exports = {
  root: true,
  overrides: [
    {
      files: ['**/*.ts'],
      parserOptions: {
        project: ['tsconfig.json'],
        tsconfigRootDir: __dirname,
        createDefaultProgram: true,
      },
      extends: [
        'plugin:@typescript-eslint/recommended',
        'plugin:@angular-eslint/recommended',
        // This is required if you use inline templates in Components
        'plugin:@angular-eslint/template/process-inline-templates',
        '@aws-amplify/amplify-ui',
      ],
      rules: {
        /*
         * `prefer-nullish-coalescing` requires `strictNullChecks` to be on.
         * Turn this off until we turn `strict` on in the `tsconfig.json`.
         */
        '@typescript-eslint/prefer-nullish-coalescing': 'off',

        // we use `amplify-` prefix for directives.
        '@angular-eslint/directive-selector': [
          'error',
          {
            type: 'attribute',
            prefix: 'amplify',
            style: 'camelCase',
          },
        ],
      },
    },
    {
      files: ['**/*.html'],
      extends: ['plugin:@angular-eslint/template/recommended'],
    },
    {
      extends: ['@aws-amplify/amplify-ui/jest'],
      files: ['**/__mocks__/**', '**/__tests__/**'],
    },
  ],
};
