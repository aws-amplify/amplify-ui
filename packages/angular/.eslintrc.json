{
  "root": true,
  "ignorePatterns": ["projects/**/*"],
  "overrides": [
    {
      "files": ["*.ts"],
      "parserOptions": {
        "project": ["tsconfig.json"],
        "createDefaultProgram": true
      },
      "extends": [
        "plugin:@angular-eslint/recommended",
        "plugin:@angular-eslint/template/process-inline-templates"
      ],
      "rules": {
        "@angular-eslint/directive-selector": [
          "error",
          {
            "type": "attribute",
            "prefix": "amplify",
            "style": "camelCase"
          }
        ]
      }
    },
    {
      "files": ["*.html"],
      "extends": ["plugin:@angular-eslint/template/recommended"],
      "rules": {}
    }
  ]
}
