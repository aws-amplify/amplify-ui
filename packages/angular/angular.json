{
  "$schema": "../../node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "ui-angular": {
      "projectType": "library",
      "root": "projects/ui-angular",
      "sourceRoot": "projects/ui-angular/src",
      "prefix": "amplify",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:ng-packagr",
          "options": {
            "tsConfig": "projects/ui-angular/tsconfig.lib.json",
            "project": "projects/ui-angular/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "projects/ui-angular/tsconfig.lib.prod.json"
            }
          }
        },
        "lint": {
          "builder": "@angular-eslint/builder:lint",
          "options": {
            "lintFilePatterns": [
              "projects/ui-angular/**/*.ts",
              "projects/ui-angular/**/*.html"
            ]
          }
        }
      }
    }
  },
  "cli": {
    "schematicCollections": [
      "@angular-eslint/schematics"
    ],
    "analytics": false
  }
}
