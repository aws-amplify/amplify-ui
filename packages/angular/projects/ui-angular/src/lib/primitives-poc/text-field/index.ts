export * from './text-field.module';
export * from './text-field-component/text-field.component';
export * from './text-field-directives/input.directive';
export * from './text-field-directives/description-text.directive';
export * from './text-field-directives/error.directive';
export * from './text-field-directives/label.directive';
