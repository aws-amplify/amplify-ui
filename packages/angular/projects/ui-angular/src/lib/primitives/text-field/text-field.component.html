<label
  class="amplify-label"
  [class.amplify-visually-hidden]="labelHidden"
  [for]="fieldId"
>
  {{ label }}
</label>
<input
  class="amplify-input"
  autocapitalize="off"
  [id]="fieldId"
  [type]="type"
  [name]="name"
  [placeholder]="placeholder"
  [required]="required"
  [value]="initialValue"
  [autocomplete]="autocomplete"
  [attr.disabled]="disabled ? '' : null"
  [attr.aria-invalid]="hasError ? 'true' : 'false'"
  [attr.aria-describedby]="describedBy"
/>
