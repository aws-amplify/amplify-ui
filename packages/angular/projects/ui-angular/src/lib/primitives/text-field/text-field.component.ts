import { Component, HostBinding, Input } from '@angular/core';
import { nanoid } from 'nanoid';

@Component({
  selector: 'amplify-text-field',
  templateUrl: './text-field.component.html',
})
export class TextFieldComponent {
  @Input() autocomplete = 'new-password';
  @Input() disabled = false;
  @Input() fieldId = `amplify-field-${nanoid(12)}`;
  @Input() initialValue = '';
  @Input() label = '';
  @Input() name: string;
  @Input() placeholder = '';
  @Input() required = true;
  @Input() type: string;
  @Input() labelHidden = false;
  @Input() hasError: boolean;
  @Input() describedBy: string;

  @HostBinding('style.display') display = 'contents';
}
