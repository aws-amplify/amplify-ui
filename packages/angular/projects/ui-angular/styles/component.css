[data-amplify-authenticator] {
  font-size: 1rem;
  line-height: 1.5rem;
  --tw-text-opacity: 1;
  color: rgba(55, 65, 81, var(--tw-text-opacity));
  --tw-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.1), 0 1px 2px 0 rgba(0, 0, 0, 0.06);
  box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000),
    var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);
  font-family: 'Work Sans', Arial, Helvetica, sans-serif;
}

[data-amplify-wrapper] [data-amplify-heading] {
  margin-bottom: 0px;
  font-size: 1.25rem;
  line-height: 1.75rem;
}

[data-amplify-authenticator] [data-amplify-form] {
  display: grid;
  --tw-bg-opacity: 1;
  background-color: rgba(255, 255, 255, var(--tw-bg-opacity));
  padding: 2rem;
}

[data-amplify-wrapper] [data-amplify-label] > :not([hidden]) ~ :not([hidden]) {
  --tw-space-y-reverse: 0;
  margin-top: calc(0.5rem * calc(1 - var(--tw-space-y-reverse)));
  margin-bottom: calc(0.5rem * var(--tw-space-y-reverse));
}

[data-amplify-confirmationcode] {
  display: grid;
}

[data-amplify-wrapper] [data-amplify-password] [data-amplify-box] {
  display: flex;
  flex-direction: row;
}

[data-amplify-wrapper]
  [data-amplify-password]
  [data-amplify-box]
  > :not([hidden])
  ~ :not([hidden]) {
  --tw-space-x-reverse: 0;
  margin-right: calc(0.25rem * var(--tw-space-x-reverse));
  margin-left: calc(0.25rem * calc(1 - var(--tw-space-x-reverse)));
}

[data-amplify-wrapper]
  [data-amplify-password]
  [data-amplify-box]
  [data-amplify-text] {
  font-size: 0.875rem;
  line-height: 1.25rem;
  --tw-text-opacity: 1;
  color: rgba(107, 114, 128, var(--tw-text-opacity));
}

[data-amplify-wrapper]
  [data-amplify-password]
  [data-amplify-box]
  [data-amplify-button] {
  font-size: 0.875rem;
  line-height: 1.25rem;
  --tw-text-opacity: 1;
  color: rgba(59, 130, 246, var(--tw-text-opacity));
}

[data-amplify-wrapper] [data-amplify-footer] [data-amplify-text] {
  font-size: 0.875rem;
  line-height: 1.25rem;
  --tw-text-opacity: 1;
  color: rgba(107, 114, 128, var(--tw-text-opacity));
}

[data-amplify-wrapper]
  [data-amplify-footer]
  [data-amplify-button]:first-of-type {
  font-size: 0.875rem;
  line-height: 1.25rem;
  --tw-text-opacity: 1;
  color: rgba(59, 130, 246, var(--tw-text-opacity));
}

[data-amplify-wrapper]
  [data-amplify-footer]
  [data-amplify-button]:last-of-type {
  border-radius: 0.25rem;
  --tw-bg-opacity: 1;
  background-color: rgba(55, 65, 81, var(--tw-bg-opacity));
  padding-left: 2rem;
  padding-right: 2rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  font-size: 1rem;
  line-height: 1.5rem;
  --tw-text-opacity: 1;
  color: rgba(255, 255, 255, var(--tw-text-opacity));
}

[data-amplify-authenticator] :disabled {
  opacity: 0.5;
  cursor: initial;
}

[data-ui-error] {
  color: orangered;
}

[data-amplify-wrapper] [data-amplify-federated] {
  display: flex;
  width: 100%;
  flex-direction: column;
  gap: 1rem;
}

[data-amplify-wrapper] [data-amplify-federated] [data-amplify-button] {
  border-radius: 0.5rem;
  background-color: transparent;
  padding: 0.5rem;
  font-size: 0.875rem;
  line-height: 1.25rem;
  --tw-text-opacity: 1;
  color: rgba(59, 130, 246, var(--tw-text-opacity));
}

[data-amplify-button] {
  cursor: pointer;
}

[data-amplify-strike] {
  width: 100%;
  text-align: center;
  border-bottom: 1px solid #c4c4c4;
  line-height: 0.1em;
  margin: 32px 0;
  color: #828282;
}

[data-amplify-strike] [data-amplify-spacer] {
  background: white;
  padding: 0 25px;
  font-size: 14px;
  font-weight: 500;
}

[data-amplify-wrapper] [data-amplify-verify-label] {
  display: flex;
}

[data-amplify-wrapper] [data-amplify-verify-input] {
  width: initial;
}

.federated-sign-in-button {
  border-color: black;
  border-radius: 0;
  color: black;
}

.federated-sign-in-button-row {
  position: relative;
}

.federated-sign-in-icon {
  position: absolute;
  left: 0;
}
