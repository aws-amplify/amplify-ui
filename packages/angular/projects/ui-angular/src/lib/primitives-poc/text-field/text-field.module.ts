import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { DescriptionDirective } from './text-field-directives/description-text.directive';
import { ErrorDirective } from './text-field-directives/error.directive';
import { InputDirective } from './text-field-directives/input.directive';
import { LabelDirective } from './text-field-directives/label.directive';
import { AmplifyTextFieldsComponent } from './text-field-component/text-field.component';

let Components: any[] = [
  AmplifyTextFieldsComponent,
  InputDirective,
  DescriptionDirective,
  ErrorDirective,
  LabelDirective,
];
@NgModule({
  declarations: Components,
  imports: [CommonModule],
  exports: Components,
})
export class AmplifyTextFieldModule {}
