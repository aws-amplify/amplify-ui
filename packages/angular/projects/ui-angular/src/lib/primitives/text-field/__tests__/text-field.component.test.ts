import { render, screen } from '@testing-library/angular';

import { TextFieldComponent } from '../text-field.component';

describe('amplify-text-field', () => {
  it('renders as expected', async () => {
    const { container } = await render(TextFieldComponent, {
      componentProperties: {
        label: 'Input',
        fieldId: 'input',
        placeholder: 'Enter something',
        type: 'text',
      },
    });
    expect(container).toMatchSnapshot();
  });

  it('should render input with autocapitalize disabled by default', async () => {
    await render(TextFieldComponent, {
      componentProperties: { name: 'username', label: 'Username' },
    });
    const input = screen.getByLabelText('Username');
    expect(input).toHaveAttribute('autocapitalize', 'off');
  });
});
