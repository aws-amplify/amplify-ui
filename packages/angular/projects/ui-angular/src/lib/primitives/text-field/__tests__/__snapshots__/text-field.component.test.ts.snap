// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-text-field renders as expected 1`] = `
<div
  id="root0"
  style="display: contents;"
>
  <label
    class="amplify-label"
    for="input"
  >
     Input

  </label>
  <input
    aria-invalid="false"
    autocapitalize="off"
    autocomplete="new-password"
    class="amplify-input"
    id="input"
    name="undefined"
    placeholder="Enter something"
    required=""
    type="text"
  />
</div>
`;
